packages/core/src/tools/error.types.ts

```diff
@@ -0,0 +1,47 @@
+import type { ClocksState } from './timeUtils'
+
+export interface ErrorWithCause extends Error {
+  cause?: Error
+}
+
+export type RawErrorCause = {
+  message: string
+  source: string
+  type?: string
+  stack?: string
+}
+
+export interface RawError {
+  startClocks: ClocksState
+  message: string
+  type?: string
+  stack?: string
+  source: ErrorSource
+  originalError?: unknown
+  handling?: ErrorHandling
+  handlingStack?: string
+  causes?: RawErrorCause[]
+}
+
+export const ErrorSource = {
+  AGENT: 'agent',
+  CONSOLE: 'console',
+  CUSTOM: 'custom',
+  LOGGER: 'logger',
+  NETWORK: 'network',
+  SOURCE: 'source',
+  REPORT: 'report',
+} as const
+
+export const NonErrorPrefix = {
+  UNCAUGHT: 'Uncaught',
+  PROVIDED: 'Provided',
+} as const
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ you could use an `enum` here